scipy/linalg/tests/test_decomp.py

```diff
@@ -404,6 +404,36 @@ def test_empty(self, dt):
         assert vr.shape == (0, 0)
         assert vr.dtype == vr_n.dtype
 
+    @pytest.mark.parametrize("include_B", [False, True])
+    @pytest.mark.parametrize("left", [False, True])
+    @pytest.mark.parametrize("right", [False, True])
```

Comment:
I need to play with `overwrite_a`/`overwrite_b` manually - I haven't thought carefully about them yet, but it might be possible to make them work for the N-dimensional arrays in some cases. That might just have to wait for lower-level implementation, though.

Reply:
I'd guess that working in-place on a batch needs pretty esoteric array strides: the last two dimensions need to be F-ordered:

```
In [12]: from scipy.linalg import inv

In [13]: a = np.arange(9).reshape(3,3) + np.eye(3)

In [14]: np.may_share_memory(a, inv(a, overwrite_a=True))
Out[14]: False

In [15]: np.may_share_memory(a, inv(a.T, overwrite_a=True))
Out[15]: True

```

now for a batch

```
In [26]: b = a.T * np.arange(1, 5)[:, None, None]   # note a.T

In [27]: b.strides
Out[27]: (72, 8, 24)

In [28]: np.may_share_memory(b, inv(b[1, ...], overwrite_a=True))
Out[28]: True

```